crates/driver/src/domain/competition/mod.rs

```diff
@@ -390,6 +477,196 @@ impl Competition {
         Ok(score)
     }
 
+    pub async fn pod_solution_submission(
+        pod_auction_client: &AuctionClient,
+        auction: &Auction,
+        deadline: chrono::DateTime<chrono::Utc>,
+        score: Option<Solved>,
+        solver: &Solver,
+    ) -> Result<(), anyhow::Error> {
+        let pod_auction_id = match u64::try_from(auction.id().map(|id| id.0).unwrap_or_default()) {
+            Ok(id) => pod_sdk::U256::from(id),
+            Err(e) => {
+                tracing::error!(error = %e, "[pod] invalid auction id");
+                return Err(anyhow::Error::new(e));
+            }
+        };
+
+        let solve_response = dto::SolveResponse::new(score.clone(), solver);
```

Comment:
Note to myself: we need to think about an upgrade path in case we have to change the response format in the future. Likely the same way ethereum handles hardforks (pick some auction_id at which all nodes should use the new format).